datalad/support/gitrepo.py

```diff
@@ -2624,8 +2609,23 @@ def get_changed_files(self, staged=False, diff_filter='', index_file=None):
             opts.append('--diff-filter=%s' % diff_filter)
         if index_file:
             kwargs['env'] = {'GIT_INDEX_FILE': index_file}
-        return [normpath(f)  # Call normpath to convert separators on Windows.
-                for f in self.repo.git.diff(*opts, **kwargs).split('\0') if f]
+        if files:
```

Comment:
surprised that test didn't fail (locally) -- I intended to make it `if files is not None` here and I thought I added explicit test (may be died in intermediate RF), TODO

Reply:
done now in db0a844c691d6848ca9627e567e658e9f3722594